ibis/expr/api.py

```diff
@@ -2400,3 +2402,183 @@ def least(*args: Any) -> ir.Value:
 
     """
     return ops.Least(args).to_expr()
+
+
+# TODO: Move
+# - _arrow_record_batch_iterator_from_iceberg_table()
+# - _arrow_batch_reader_from_iceberg_table()
+# somewhere else.
+def _arrow_record_batch_iterator_from_iceberg_table(
+    iceberg_table: pyiceberg.table.Table,
+    snapshot_id: str = None,
+    options: dict[str, str] = {},
+    limit: int = None,
+) -> Iterator[pa.RecordBatch]:
+    # Note: This function is a modification of `pyiceberg.table.to_arrow()`.
+    # The modification has been done to read the Iceberg table as RecordBatch
+    # iterator rather than a single Arrow table.
+
+    from pyiceberg.expressions import AlwaysTrue
+    from pyiceberg.io.pyarrow import _task_to_table, PyArrowFileIO
+
+    scan_result = iceberg_table.scan(
+        snapshot_id=snapshot_id,
+        options=options,
+        limit=limit,
+    )
+
+    scheme, netloc, _ = PyArrowFileIO.parse_location(iceberg_table.location())
+    if isinstance(iceberg_table.io, PyArrowFileIO):
+        fs = iceberg_table.io.fs_by_scheme(scheme, netloc)
+    else:
+        try:
+            from pyiceberg.io.fsspec import FsspecFileIO
```

Comment:
It will only fall back to `Fsspec` if `PyArrow` is not installed, but I don't think that's the case with Ibis.